src/main/java/com/example/demo/controllers/LocationController.java

```diff
@@ -0,0 +1,38 @@
+package com.example.demo.controllers;
+
+import com.example.demo.entities.Location;
+import com.example.demo.services.LocationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.web.bind.annotation.*;
+
+@RestController("/api")
+@RequestMapping(value = "/api")
+public class LocationController {
+
+    @Autowired
+    LocationService locationService;
+
+    @PostMapping("/locations")
```

Comment:
Again, probably going to want to be explicit about json
```suggestion
    @PostMapping(value = "/locations", produces = MediaType.APPLICATION_JSON, consumes = MediaType.APPLICATION_JSON)
```